### ISSUE TITLE
-Wunneeded-internal-declaration in drivers/clk/clk-s2mps11.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#228]
```
drivers/clk/clk-s2mps11.c:242:34: warning: variable 's2mps11_dt_match' is not needed and will not be emitted [-Wunneeded-internal-declaration]
static const struct of_device_id s2mps11_dt_match[] = {
                                 ^
1 warning generated.
```

Patch sent: https://lore.kernel.org/lkml/20181018191340.30440-1-natechancellor@gmail.com/

**Comment by nathanchance** [comment]
Patch accepted: https://git.kernel.org/pub/scm/linux/kernel/git/clk/linux.git/commit/?id=9c940bbe2bb47e03ca5e937d30b6a50bf9c0e671

**Comment by nathanchance** [comment]
Merged into mainline: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=9c940bbe2bb47e03ca5e937d30b6a50bf9c0e671

**Comment by nickdesaulniers** [comment]
possibly also fixed on the llvm side by #232
